File: app/src/main/java/au/id/blackwell/kurt/lantv/player/TvPlayer.java
package au.id.blackwell.kurt.lantv.player;

import android.view.View;

import au.id.blackwell.kurt.lantv.resolver.MediaResolver;

public interface TvPlayer {
    /**
     * Reset player, and ready new media to load
     * @param mediaResolver
     */
    void reset(MediaResolver mediaResolver);

    /**
     * Resume playing the video
     */
    void play();

    /**
     * Pause video, if playing
     */
    void pause();

    /**
     * Stop playing the video.
     */
    void stop();

    /**
     * Set a listener to be told of the playback status
     * @param listener
     */
    void setTvPlayerListener(TvPlayerStatusListener listener);

    /**
     * Get the view for this player.
     */
    View getView();

    /**
     * Release resources from player
     */
    void release();
}

File: app/src/main/java/au/id/blackwell/kurt/lantv/player/TvPlayer.kt
package au.id.blackwell.kurt.lantv.player

import android.view.View

import au.id.blackwell.kurt.lantv.resolver.MediaResolver

interface TvPlayer {

    /**
     * Get the view for this player.
     */
    val view: View

    /**
     * Reset player, and ready new media to load
     * @param mediaResolver
     */
    fun reset(mediaResolver: MediaResolver)

    /**
     * Resume playing the video
     */
    fun play()

    /**
     * Pause video, if playing
     */
    fun pause()

    /**
     * Stop playing the video.
     */
    fun stop()

    /**
     * Set a listener to be told of the playback status
     * @param listener
     */
    fun setTvPlayerListener(listener: TvPlayerStatusListener?)

    /**
     * Release resources from player
     */
    fun release()
}
